source/server/connection_handler_impl.cc

```diff
@@ -172,6 +172,7 @@ Network::BalancedConnectionHandlerOptRef
 ConnectionHandlerImpl::getBalancedHandlerByAddress(const Network::Address::Instance& address) {
   // This is a linear operation, may need to add a map<address, listener> to improve performance.
   // However, linear performance might be adequate since the number of listeners is small.
+  // If this is modified in the future, note that a listener's address may change.
```

Comment:
Was worried this optimization could break if a listener's address is changed, but actually should probably just remove this comment... `ConnectionHandlerImpl`'s listeners' addresses wouldn't change; instead a new listener would be added with `addListener`, and once the old one is drained it's removed with `removeListeners`